src/vm.c

```diff
@@ -2662,6 +2662,9 @@ mrb_vm_exec(mrb_state *mrb, const struct RProc *proc, const mrb_code *pc)
       if (mrb_obj_eq(mrb, regs[a], regs[a+1])) {
         SET_TRUE_VALUE(regs[a]);
       }
+      else if (mrb_type(regs[a]) == MRB_TT_SYMBOL) {
```

Comment:
Thank you for the pull-request. This is a nice idea. But if the condition is `mrb_immediate_p(regs[a])`, equalify comparison will be (slightly) faster not only for symbols, but other immediate values such as integers and float numbers.

Reply:
this unfortunately breaks some test cases, as it means that some (e.g. `Integer#==(Rational)`) equality checks are never performed.